### PR TITLE
feat: undefined props value should result in undefined select state

### PR DESCRIPTION
**Pull request by iambumblehead**
cc @lwhiteley @rtyurin

### CODE REVIEW

**Comment by lwhiteley** [comment]
This enables you to clear the dropdown programmatically which is good but should propably also add a boolean prop.

- [ ] maybe `isClearable` that adds a clearing option to the dropdown list
- [ ] maybe add a storybook example showcasing this as well

**Comment by iambumblehead** [comment]
I can't think of a situation where the parent component set the value to `undefined` but expects the dropdown internal state to be different. Is an `isClearable` property necessary? Would you elaborate on that?

**Comment by lwhiteley** [comment]
best example i can give is react-select

https://react-select.com/advanced#advanced

**Comment by iambumblehead** [comment]
it seems like we would want the storybook entry to have,
 * a boolean/checkbox component for setting `isClearable` and
 * a button inside either the storybook canvas or the storybook controls area for 'clearing' the dropdown value

Adding the boolean checkbox is straightforward but I'm not sure how to add the control for 'clearing' the value now. Maybe the 'clear' button would be a checkbox that always remains in the off/false state... not sure. Do you have an implementation suggestion @lwhiteley?

**Comment by lwhiteley** [comment]
maybe prioritize 
> a boolean/checkbox component for setting isClearable and

and leave this one for later
> a button inside either the storybook canvas or the storybook controls area for 'clearing' the dropdown value

im still also figuring out how to do controlled examples so far not seeing much good examples in the wild

**Comment by iambumblehead** [comment]
I'll update the branch tomorrow sometime

**Comment by iambumblehead** [comment]
@lwhiteley a `[clear]` button is added to the canvas. If you have any thoughts about it or even if you want to remove it we can keep making changes here.

**Comment by iambumblehead** [comment]
>> "maybe isClearable that adds a clearing option to the dropdown list"

Do you think a separate "clear" option or button should be provided by the dropdown, inside the dropdown list? What do you think should be added here?

**Comment by lwhiteley** [comment]
Yea I think it should be added to the drop-down. Either added to the top of
the list or on the selector beside the drop-down arrow. Whichever makes
more sense

**Comment by iambumblehead** [comment]
@lwhiteley ok

**Comment by iambumblehead** [comment]
@lwhiteley I'll try to get _something_ up here today. There are a few details that could need tuning and maybe a few follow-up changes will be needed. A few questions that pop up,
 * the vertical-line separator between the clear button and the arrow button used by react-select is a nice touch --do we want that as well?
 * should the clear icon have a hover state? the arrow icon does not have one.
 * if the clear icon 'x' is squarely proportioned and the up/down icon is rectangular proportion, will the icons look misaligned when they are side by side?
 * should a custom clear icon be supported?
 * multiple situations to test and handle: custom close icon, hiding the close icon when no selection is present, making the big-enough hover area for the close icon, etc.

I'll try to get answer these to the best of my ability and get something up here to review

**Comment by iambumblehead** [comment]
Screen Shot 2020-11-01 at 19 14 21

Here's a screenshot of what we have so far. I'll update things so the tests and pipeline pass.

**Comment by iambumblehead** [comment]
what do you think of these changes @lwhiteley ?
 * the storybook [clear] button is removed,
 * a clear button is added inside the dropdown,
 * clear button only appears when value is selected

File: src/stories/Dropdown.stories.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ FlatArrayExample.args = {
   onChange: (option) => {
     console.log('You selected ', option.label);
   },
+  isClearable: true,
```
Comment (lwhiteley) [inline src/stories/Dropdown.stories.js:34]: create an example specifically for clearing the value

File: src/stories/Dropdown.stories.js [hunk 2]
```diff
@@ -86,6 +87,16 @@ customArrowExample.args = {
   options: ['one', 'two', 'three'],
 };
 
+export const customClearExample = Template.bind({});
+const clearIcon = <span className="clear">X</span>;
+
+customClearExample.args = {
```
Comment (lwhiteley) [inline src/stories/Dropdown.stories.js:93]:
```suggestion
export const customClearIconExample = Template.bind({});
const clearIcon = <span className="clear">X</span>;

customClearIconExample.args = {
```

File: src/Dropdown.js [hunk 3]
```diff
@@ -25,7 +26,8 @@ function Dropdown({
   className,
   noOptionsDisplay,
   innerRef,
-  menu: MenuContainer
+  menu: MenuContainer,
+  isClearable = true
```
[the remark this answers is not in the document]
Comment (lwhiteley) [reply src/Dropdown.js:31]: also need to add the new props to the typescript file

[the remark this answers is not in the document]
Comment (iambumblehead) [reply src/Dropdown.js:31]: my ts file updates might be incorrect, feel free to open another issue for that

**Comment by iambumblehead** [comment]
@lwhiteley the commit-suggestion didn't pass the linter :/

File: src/stories/Dropdown.stories.js (hunk 1 above)
Comment (iambumblehead) [reply src/stories/Dropdown.stories.js]: a story is added for the clear button

**Comment by lwhiteley** [comment]
> @lwhiteley the commit-suggestion didn't pass the linter :/

amend the commit title

**Comment by iambumblehead** [comment]
the shell on this machine crashes when git goes into interactive mode, so I'll try to do the amend tomorrow

**Comment by iambumblehead** [comment]
I updated git to use gui-emacs so that it wouldn't crash and that got it

File: src/Dropdown.js [hunk 4]
```diff
@@ -25,7 +26,9 @@ function Dropdown({
   className,
   noOptionsDisplay,
   innerRef,
-  menu: MenuContainer
+  menu: MenuContainer,
+  clearIcon,
+  isClearable = false
```
Comment (lwhiteley) [inline src/Dropdown.js:31]: probably can jus be 

```js
isClearable,
```

**Review by lwhiteley — APPROVED** [review]

**Pull request merged**